galaxy_ng/app/api/views.py

```diff
@@ -25,7 +26,10 @@
 
 
 class ApiRootView(api_base.APIView):
-    permission_classes = [IsAuthenticated]
+    if settings.GALAXY_ENABLE_UNAUTHENTICATED_COLLECTION_DOWNLOAD:
+        permission_classes = []
+    else:
+        permission_classes = [IsAuthenticated]
```

Comment:
This cannot be done at the class level, because this `if` will be evaluated at the class definition time and it is better for permissions like this to be evaluated at runtime, so it is also easier for testing.

You have to create a new Access Policy `ApiRootAccessPolicy` and inherit and/or implement the same idea of this other one here https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/access_control/access_policy.py#L75-L76

You can see the mixin here https://github.com/ansible/galaxy_ng/blob/6ce0f25b4fd5b93d778099dfe23bbf3f9f26d44d/galaxy_ng/app/access_control/mixins.py#L41

then on the permissions you can set like this https://github.com/ansible/galaxy_ng/blob/6ce0f25b4fd5b93d778099dfe23bbf3f9f26d44d/galaxy_ng/app/access_control/statements/standalone.py#L12

@bmclaughlin implemented all of this and might be able to give you some hints.

Reply:
This is the approach I'd go with as well.

Reply:
I tried to implement this, but I am wondering, is this okey for the other AppRootViewSet using endpoints too? I feel like I should create a new ViewSet just for the `content/` and `content/.../api` endpoint (and redirect endpoint)